helpers/query.go

```diff
@@ -42,16 +43,18 @@ func QueryBalance(chain *relayer.Chain, address string, showDenoms bool) (sdk.Co
 			continue
 		}
 
-		for i, d := range dts.DenomTraces {
-			if c.Denom == d.IBCDenom() {
-				out = append(out, sdk.Coin{Denom: d.GetFullDenomPath(), Amount: c.Amount})
-				break
-			}
+		// splitting denom to get hash
+		denomSplit := strings.Split(c.Denom, "/")
```

Comment:
I don't think we can assume an account only has IBC tokens. We need to check if the prefix of the coin is `ibc/`. I added an issue on [`ibc-go`](https://github.com/cosmos/ibc-go/issues/102) since there should be a helper function for this

It also seems odd to me that we need to parse out `ibc` before doing the QueryDenomTrace, so I added another [issue](https://github.com/cosmos/ibc-go/issues/101)

I'm okay actually closing this issue and leaving the code as is until we fix the above issues upstream. I'd like a function that looks something like this:
```go
for _, c := range coins {
    if IsIBCDenom(c.Denom) {
        dt, err := chain.QueryDenomTrace(c.Denom)
        if err != nil {
            return fmt.Errorf("unregistered token in balance")
        }
        
        out = append(out, NewToken(dt.DenomTrace.GetFullDenomPath(), c.Amount)
    } else {
         out = append(out, c)
    }
}
```